### ISSUE TITLE
yolov3-tiny createEngine failed

### ISSUE THREAD

**Issue by arnoldfychen** [wang-xinyu/tensorrtx#643]
Following the README of https://github.com/wang-xinyu/tensorrtx/tree/master/yolov3-tiny,
I downloaded wget -c https://pjreddie.com/media/files/yolov3-tiny.weights and converted it with gen_wts.py to yolov3-tiny.wts, and built out the binary yolov3-tiny under ./build and executed  ./yolov3-tiny -s,  always failed in createEngine(),  the error message is :

[07/23/2021-17:39:03] [E] [TRT] ../builder/Network.cpp (1262) - Assertion Error in updateTensor: 0 (isValidDims(out->form.dims, hasImplicitBatchDimension(), false))

this error actually happened at this statement:

ICudaEngine* engine = builder->buildEngineWithConfig(*network, *config);

so, builder->buildEngineWithConfig() always returned a NULL engine. 

My TensorRT version is 7.1.3.0-1+cuda10.2 .

**Comment by wang-xinyu** [comment]
Did you use the archive branch of yolov3? We only support archive branch.

**Comment by arnoldfychen** [comment]
Yes, I did use this command:  git clone -b archive https://github.com/ultralytics/yolov3.git

**Comment by wang-xinyu** [comment]
@arnoldfychen Can you paste the complete error msg?

**Comment by arnoldfychen** [comment]
Loading weights: ./yolov3-tiny.wts
outputDims: nb:3,[48,13,13,-1683795936]

Building engine, please wait for a while...
[07/26/2021-10:38:10] [E] [TRT] ../builder/Network.cpp (1262) - Assertion Error in updateTensor: 0 (isValidDims(out->form.dims, hasImplicitBatchDimension(), false))
Build engine successfully!
==================== engine created: 0

yolov3-tiny: /home/jzyq/tensorrtx/yolov3-tiny/yolov3-tiny.cpp:324: void APIToModel(unsigned int, nvinfer1::IHostMemory**): Assertion `engine != nullptr' failed.
Aborted (core dumped)

These are all the messages, please note I added some my own printings in yolov3-tiny.cpp, so, 'Assertion `engine != nullptr' failed' happened at yolov3-tiny.cpp:324,  the assertion is really here:

     ICudaEngine* engine = createEngine(maxBatchSize, builder, config, DataType::kFLOAT);
     [assert(engine](url) != nullptr);

**Comment by wang-xinyu** [comment]
What else did you change? Can you try the default code?

**Comment by arnoldfychen** [comment]
I printed out the output dimensions of layer15 and layer22:

conv15 outputDims: nb:3,[48,13,13,-1801150064]
conv22 outputDims: nb:3,[48,26,26,-1]

they are right, it look like the cause of the assertion error 'Assertion Error in updateTensor: 0'  is in yolo plugin:

    auto creator = getPluginRegistry()->getPluginCreator("YoloLayer_TRT", "1");
    const PluginFieldCollection* pluginData = creator->getFieldNames();
    IPluginV2 *pluginObj = creator->createPlugin("yololayer", pluginData);
    ITensor* inputTensors_yolo[] = {conv15->getOutput(0), conv22->getOutput(0)}; 
    auto yolo = network->addPluginV2(inputTensors_yolo, 2, *pluginObj);

 don't know where this plugin is not correctly implemented so that isValidDims(out->form.dims, hasImplicitBatchDimension(), false) returned false in TensorRT's Network.cpp.   I guess TensorRT 7.1.3 might have added some new constraints, while the version you had ever used for testing this yolov3-tiny doesn't have those constraints.

**Comment by arnoldfychen** [comment]
@wang-xinyu  I did nothing but only added some printings in yolov3-tiny.cpp for investigating this crash.

**Comment by wang-xinyu** [comment]
Did you change the input size? It should not be 13x13, if using 608.

**Comment by arnoldfychen** [comment]
@wang-xinyu  Yes, I only changed the following parameters as per our yolov3-tiny model

    static constexpr int CLASS_NUM = 11; //80;
    static constexpr int INPUT_H = 416; //608;
    static constexpr int INPUT_W = 416; //608;

**Comment by wang-xinyu** [comment]
So try official model first, with default code.

**Comment by arnoldfychen** [comment]
@wang-xinyu    I downloaded the yolov3-tiny.zip weights file from https://pan.baidu.com/s/19s6hO8esU7-TtZEXN7G3OA which you shared, and built out binary from your original code without any change,  and run './yolov3-tiny -s', still got the same crash, so, this crash has nothing with the class number of input shape.

**Comment by wang-xinyu** [comment]
I just tried yolov3-tiny.zip and master branch latest commit from my end, T4, trt7.2.3, cuda10.2. It works fine.
So probably your environment not right.

**Comment by arnoldfychen** [comment]
I work on Jetson Nano with JetPack 4.3 (cuda10.2, TensorRT 7.1.3 ), not on PC

**Comment by arnoldfychen** [comment]
I also ran into this trouble when I was integrating yolov5-tensorrt v6 network into application yesterday, and I have managed to find the cause is my environment has a yolo*.so file preloaded by PRE_LOAD and that so file has some functions whose names are the same as the names of the functions in the so file of yolov5-tensorrt, so, when application got running, it was really using some unexpected functions, so, failed in creating engine.   
This cause is seldom seen, but I want to add this comment for helping someone who may also run into this trouble.
